koku/masu/database/report_db_accessor_base.py

```diff
@@ -153,6 +164,27 @@ def _execute_trino_raw_sql_query_with_description(
         LOG.info(log_json(msg="executed trino sql", log_ref=log_ref, running_time=running_time, context=ctx))
         return results, description
 
+    def _handle_trino_external_error(
```

Comment:
This would be better as a [decorator](https://realpython.com/primer-on-python-decorators/). That way we would not need separate but mostly similar implementations and we avoid any potential issues that arise from recursive calls.

[Here is a pretty good example](https://github.com/pyprogrammerblog/python-retry/blob/master/python_retry/retry.py#L9).

Reply:
I've created a decorator. Thanks!